assets/js/list.js

```diff
@@ -1,14 +1,63 @@
-const books = [
-  {
-    title: 'title1',
-    author: 'Didy',
-  },
-  {
-    title: 'title1',
-    author: 'Sylvester',
-  },
-  {
-    title: 'title1',
-    author: 'Raphael',
-  },
-];
\ No newline at end of file
+let books = JSON.parse(localStorage.getItem('bookData')) || [];
+
+const addButton = document.querySelector('#addBook');
+const formTitle = document.getElementById('title');
+const formAuthor = document.getElementById('author');
+const bookList = document.querySelector('#list');
+
+function addBook(title, author) {
+  if (title !== '' && author !== '') {
+    const book = {
+      id: books.length + 1,
+      title,
+      author,
+    };
+
+    books.push(book);
+    localStorage.setItem('bookData', JSON.stringify(books));
+  }
+}
+
+function removeBook(id) {
+  let bookData = JSON.parse(localStorage.getItem('bookData'));
+  bookData = bookData.filter((local) => local.id !== parseInt(id, 10));
+  books = bookData;
+  localStorage.setItem('bookData', JSON.stringify(books));
+}
+
+function displaylist() {
+  books.forEach((book) => {
+    const bookCard = document.createElement('li');
+    const removeButton = document.createElement('button');
+    const hr = document.createElement('hr');
+
+    bookCard.innerHTML += `    
+    <p>${book.title}</p>
+    <p>${book.author}</p>  
+       `;
+    bookCard.id = `data-${book.id}`;
+    removeButton.textContent = 'Remove';
+    removeButton.dataset.id = book.id;
+    removeButton.addEventListener('click', (e) => {
+      const { id } = e.target.dataset;
+      removeBook(id);
+      const bookEl = document.getElementById(`data-${book.id}`);
+      bookList.removeChild(bookEl);
+      window.location.reload();
+    });
+    bookCard.appendChild(removeButton);
+    bookCard.appendChild(hr);
+    bookList.appendChild(bookCard);
+  });
+}
+
+addButton.addEventListener('click', (e) => {
+  e.preventDefault();
+  addBook(formTitle.value, formAuthor.value);
+
+  formTitle.value = '';
+  formAuthor.value = '';
+  window.location.reload();
+});
+
+displaylist();
```

Comment:
[OPTIONAL] Well done :+1: you did a very nice job :clap: but I suggested you some changes if I add a new book your browser is refreshed then the data is displayed so this is not a good practice so if you implemented like that if you add or remove your book then its added without refreshing then it's looking good :+1: check below :point_down: my suggestion :slightly_smiling_face: 

```suggestion
const bookList = document.querySelector('#list');
const addButton = document.querySelector('#addBook');
const formTitle = document.getElementById('title');
const formAuthor = document.getElementById('author');

let books = [];

function addBook(title, author) {
  if (title === '' || author === '') return;

  const book = { id: books.length + 1, title, author };
  books.push(book);
  localStorage.setItem('bookData', JSON.stringify(books));
}

function removeBook(id) {
  books = books.filter((book) => book.id !== parseInt(id, 10));
  localStorage.setItem('bookData', JSON.stringify(books));
  displayList();
}

function displayList() {
  bookList.innerHTML = '';

  books.forEach((book) => {
    const bookCard = document.createElement('li');
    const removeButton = document.createElement('button');
    const hr = document.createElement('hr');

    bookCard.innerHTML = `
      <p>${book.title}</p>
      <p>${book.author}</p>
    `;
    bookCard.id = `data-${book.id}`;
    removeButton.textContent = 'Remove';
    removeButton.dataset.id = book.id;
    removeButton.addEventListener('click', () => {
      removeBook(book.id);
    });

    bookCard.appendChild(removeButton);
    bookCard.appendChild(hr);
    bookList.appendChild(bookCard);
  });
}

function init() {
  books = JSON.parse(localStorage.getItem('bookData')) || [];
  displayList();
}

addButton.addEventListener('click', (e) => {
  e.preventDefault();
  addBook(formTitle.value, formAuthor.value);
  formTitle.value = '';
  formAuthor.value = '';
  displayList();
});

init();
```

**_The changes made to the code are as follows:_**

- The `books` variable is now declared without retrieving it from `localStorage` in the same line. It's initialized to an empty array at first and then it's populated with data from `localStorage` in the `init` function.

- The `addBook` function now checks if the `title` or `author` values are empty before creating a new book object.

- The `removeBook` function now updates the `books` array directly instead of retrieving the data from `localStorage` and filtering it.

- The `displayList` function now clears the `bookList` element before populating it with new book items.

- The event listener for the remove button now passes the `book.id` value to the `removeBook` function instead of using the event target's `dataset` property.

- The `window.location.reload()` calls are removed from the code, so the page doesn't refresh after adding or removing a book. Instead, the `displayList` function is called to update the UI.